<launch>
    <!-- <rosparam file="$(find steer_track)/config/camera.yaml" command="load"/> -->
    <!-- <rosparam file="$(find steer_track)/config/camera2.yaml" command="load"/>
    <node pkg="steer_track" type="hk_camera_node" name="hikrobot_camera" respawn="true" output="screen" />  -->
    <!-- 
    pkg指出ROS应该运行哪个pkg
    type是可执行文件的名称
    name则是可以任意给出的，它覆盖了原文件中ros::init指定的node的名字 -->
    <rosparam file="$(find steer_track)/config/camera2.yaml" command="load"/>
    <node pkg="steer_track" type="hk_camera_node" name="hikrobot_camera1" respawn="true" output="screen" >
    <param name="hk_name"  type = "string"    value="hk_camere1"/>
    </node>

  <arg name="launch_prefix" default="" />
  <!-- configure camera input -->
  <arg name="camera_name1" default="/hikrobot_camera1" />
  <arg name="image_topic1" default="image" />
  <arg name="queue_size1" default="1" />
  <!-- apriltag_ros continuous detection node -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" ns="camera_1">
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name1)/$(arg image_topic1)" />
    <remap from="camera_info" to="$(arg camera_name1)/camera_info" />

    <param name="publish_tag_detections_image" type="bool" value="true" /><!-- default: false -->
    <param name="queue_size" type="int" value="$(arg queue_size1)" />

    <!-- load parameters (incl. tag family, tags, etc.) -->
    <rosparam command="load" file="$(find apriltag_ros)/config/settings.yaml"/>
    <rosparam command="load" file="$(find apriltag_ros)/config/tags.yaml"/>
  </node>


    <!--调用发布定位信息的文件，若使用optitrack，其中关键变量为steerRobot，需要修改为刚体的名字-->
  <arg name="climbot_name" default="steer_tk"/>
  <node pkg="steer_track" type="pose_vel_pub_node" name="pose_vel_pub_node" args="$(arg climbot_name)" output="screen" ns="robot_1"/>


</launch>
